python/semantic_kernel/connectors/ai/azure_ai_inference/services/azure_ai_inference_text_embedding.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from typing import Any
+
+from azure.ai.inference import load_client as load_client_sync
+from azure.ai.inference.aio import EmbeddingsClient as EmbeddingsClientAsync
+from azure.ai.inference.models import EmbeddingsResult, ModelInfo, ModelType
+from azure.core.credentials import AzureKeyCredential
+from numpy import array, ndarray
+from pydantic import ValidationError
+
+from semantic_kernel.connectors.ai.azure_ai_inference.azure_ai_inference_prompt_execution_settings import (
+    AzureAIInferenceEmbeddingPromptExecutionSettings,
+)
+from semantic_kernel.connectors.ai.azure_ai_inference.azure_ai_inference_settings import AzureAIInferenceSettings
+from semantic_kernel.connectors.ai.embeddings.embedding_generator_base import EmbeddingGeneratorBase
+from semantic_kernel.exceptions.service_exceptions import ServiceInitializationError
+
+
+class AzureAIInferenceTextEmbedding(EmbeddingGeneratorBase):
+    """Azure AI Inference Text Embedding Service."""
+
+    client: EmbeddingsClientAsync
+
+    def __init__(
+        self,
+        api_key: str | None = None,
+        endpoint: str | None = None,
+        service_id: str | None = None,
+        env_file_path: str | None = None,
+        env_file_encoding: str | None = None,
+        client: EmbeddingsClientAsync | None = None,
+        model_info: ModelInfo | None = None,
+    ) -> None:
+        """Initialize the Azure AI Inference Text Embedding service.
+
+        Args:
+            api_key: The API key for the Azure AI Inference service deployment.
+            endpoint: The endpoint of the Azure AI Inference service deployment.
+            service_id: Service ID for the chat completion service. (Optional)
+            env_file_path: The path to the environment file. (Optional)
+            env_file_encoding: The encoding of the environment file. (Optional)
+            client: The Azure AI Inference client to use. (Optional)
+            model_info: The model info of the provided client. (Optional, required if client is provided)
+        """
+        if client is not None:
+            if model_info is None:
+                raise ServiceInitializationError("Model info is required when providing a client.")
+        else:
```

Comment:
This is just a style comment, so feel free to disregard: if the client is none and then model_info is none, the exception won't allow the code to continue, so the `else` isn't needed here. I also think we could reduce the nested `if` statements.

Reply:
The nested conditional and the else are actually needed.

The client is allowed to be `None`, in which case we will create a client. The `model_info` is required if `client` is not `None`. 

If we have 
```Python
        if client is not None and model_info is None:
            raise ServiceInitializationError("Model info is required when providing a client.")

        try:
            azure_ai_inference_settings = AzureAIInferenceSettings.create(
                api_key=api_key,
                endpoint=endpoint,
                env_file_path=env_file_path,
                env_file_encoding=env_file_encoding,
            )
        except ValidationError as e:
            raise ServiceInitializationError(f"Failed to validate Azure AI Inference settings: {e}") from e

        client, model_info = self._create_client(azure_ai_inference_settings)
```
, it will override the `client` and `model_info` if there are not `None`.

Reply:
I understand, thanks!